File: Week 3/C++ Codes/CheckIfLinkedListIsPalindrome.cpp
#include<bits/stdc++.h>
using namespace std;

struct Node {
  int data;
  struct Node *next;
  Node(int x) {
    data = x;
    next = NULL;
  }
};

/*You are required to complete this method */
bool isPalindrome(Node *head)
{
    stack<int> s;
    struct Node *temp = head;
    while (temp != NULL) {
        s.push(temp->data);
        temp = temp->next;
    }
    while (head != NULL) {
        int num = s.top();
        s.pop();
        if (head->data != num)
            return false;
        head = head->next;
    }
    return true;
}

File: Week 3/Python Codes/CheckIfLinkedListIsPalindrome.py
# Node Class
class Node:
    def __init__(self, data):   # data -> value stored in node
        self.data = data
        self.next = None

def isPalindrome(head):
    temp = head
    l = []
    while(temp is not None):
        l.append(temp.data)
        temp = temp.next
    temp = head
    while (temp is not None):
        num = l.pop()
        if (temp.data != num):
            return False
        temp = temp.next
    return True